### PR TITLE
`sage.arith`: Update `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
- Part of: #29705 
- Cherry-picked from: #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/arith/misc.py [hunk 1]
```diff
@@ -81,109 +81,111 @@ def algdep(z, degree, known_bits=None, use_bits=None, known_digits=None,
...
         19*x - 7
 
     These examples show the importance of proper precision control. We
     compute a 200-bit approximation to `sqrt(2)` which is wrong in the
     33'rd bit::
 
+        sage: # needs sage.rings.real_mpfr
         sage: z = sqrt(RealField(200)(2)) + (1/2)^33
-        sage: p = algdep(z, 4); p                                                       # optional - sage.libs.pari
+        sage: p = algdep(z, 4); p                                                       # needs sage.libs.pari
```
Comment (dcoudert) [inline src/sage/arith/misc.py]: block scope

File: src/sage/arith/misc.py [hunk 2]
```diff
@@ -592,19 +594,19 @@ def is_pseudoprime(n):
 
     EXAMPLES::
 
-        sage: is_pseudoprime(389)                                                       # optional - sage.libs.pari
+        sage: is_pseudoprime(389)                                                       # needs sage.libs.pari
```
Comment (dcoudert) [inline src/sage/arith/misc.py]: block scope

File: src/sage/arith/misc.py [hunk 3]
```diff
@@ -1616,21 +1635,21 @@ class Sigma:
 
         sage: sigma(100,4)
         106811523
-        sage: sigma(factorial(100), 3).mod(144169)                                      # optional - sage.libs.pari
+        sage: sigma(factorial(100), 3).mod(144169)                                      # needs sage.libs.pari
```
Comment (dcoudert) [inline src/sage/arith/misc.py]: block scope ?

File: src/sage/arith/misc.py [hunk 4]
```diff
@@ -5034,26 +5068,26 @@ def falling_factorial(x, a):
 
         sage: falling_factorial(10, 3)
         720
-        sage: falling_factorial(10, RR('3.0'))                                          # optional - sage.symbolic
+        sage: falling_factorial(10, RR('3.0'))                                          # needs sage.symbolic
```
Comment (dcoudert) [inline src/sage/arith/misc.py]: block scope

File: src/sage/arith/misc.py [hunk 5]
```diff
@@ -5922,30 +5956,30 @@ def sort_complex_numbers_for_display(nums):
...
+        sage: nums = [CDF(i) for i in range(3)]                                         # needs sage.rings.complex_double
+        sage: for i in range(3):                                                        # needs sage.rings.complex_double
         ....:     nums.append(CDF(i + RDF.random_element(-3e-11, 3e-11),
         ....:                     RDF.random_element()))
         ....:     nums.append(CDF(i + RDF.random_element(-3e-11, 3e-11),
         ....:                     RDF.random_element()))
-        sage: shuffle(nums)
-        sage: nums = sort_c(nums)
-        sage: for i in range(len(nums)):
+        sage: shuffle(nums)                                                             # needs sage.rings.complex_double
```
Comment (dcoudert) [inline src/sage/arith/misc.py]: block scope

**Review by dcoudert — COMMENTED** [review]
block scope could be used in few more places. otherwise LGTM.

**Comment by mkoeppe** [comment]
Thank you, all done in 565fb2b

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**